palace/utils/geodata.cpp

```diff
@@ -1096,16 +1098,32 @@ std::unique_ptr<mfem::Mesh> LoadMesh(const std::string &path)
       MFEM_ABORT("Unable to open translated mesh file \"" << tmp << "\"!");
     }
 #endif
-    return std::make_unique<mfem::Mesh>(fi, 1, 1, true);
+    mesh = std::make_unique<mfem::Mesh>(fi, 1, 1, true);
   }
-  // Otherwise, just rely on MFEM load the mesh.
-  std::ifstream fi(path);
-  if (!fi.good())
+  else
   {
-    MFEM_ABORT("Unable to open mesh file \"" << path << "\"!");
+    // Otherwise, just rely on MFEM load the mesh.
+    std::ifstream fi(path);
+    if (!fi.good())
+    {
+      MFEM_ABORT("Unable to open mesh file \"" << path << "\"!");
+    }
+    mesh = std::make_unique<mfem::Mesh>(fi, 1, 1, true);
+  }
+  if (mesh->GetNodes() && remove_curvature)
+  {
+    mfem::GridFunction *nodes = nullptr;
+    int own_nodes = true;
+    mesh->SwapNodes(nodes, own_nodes);
+    if (own_nodes)
+    {
+      delete nodes;
+    }
```

Comment:
This works fine if the mesh is never marked as nonconformal, but I've been doing some further testing and instead of using `DistributeMesh`
```
  smesh->EnsureNCMesh(true);
  std::unique_ptr<mfem::ParMesh> mesh =
      std::make_unique<mfem::ParMesh>(comm, *smesh, partitioning.get());
```
having made sure all ranks build the mesh (even running with 1 rank), causes the eigen linear solve to fail on cavity example. Noting this here, but this is arguably downstream of this pr, so going to approve.

Reply:
Definitely worth looking into for `NCMesh` support, thanks for noting. We probably need more details on how the example "fails" with `"RemoveCurvature"` to debug.